app/src/main/java/org/cru/godtools/adapter/OnboardingPagerAdapter.kt

```diff
@@ -0,0 +1,21 @@
+package org.cru.godtools.adapter
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentPagerAdapter
+import org.cru.godtools.activity.OnBoardingCallbacks
+import org.cru.godtools.fragment.OnboardingSlideFragment
+
+class OnboardingPagerAdapter(val callbacks: OnBoardingCallbacks, fm: FragmentManager) :
+    FragmentPagerAdapter(fm) {
+
+    override fun getItem(position: Int): Fragment {
+        return OnboardingSlideFragment.newInstance(position).apply {
+            setCallback(callbacks)
+        }
+    }
+
+    override fun getCount(): Int {
+        return 4
```

Comment:
Is there no way to determine this dynamically?

Reply:
a better pattern for this is to create an enum with what the pages are and use that to drive the pager adapter. This makes it much easier to add/remove pages and reorder them in the future.

Reply:
for example: https://github.com/CruGlobal/mpdx-android/blob/develop/app/src/main/java/org/mpdx/features/appeals/details/AppealDetailsPagerAdapter.kt

because you are just loading a layout, you may even want to include the layout id within the enum